test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp

```diff
@@ -0,0 +1,175 @@
+ */
+
+// make sure the CPU_B_SEQ_T_SEQ backend is always available
+#ifndef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#    define ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#endif // ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#include <alpaka/alpaka.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+
+// fill an trivial type with std::memset
+template<typename T>
+constexpr T memset_value(int c)
+{
+    T t;
+    std::memset(&t, c, sizeof(T));
+    return t;
+}
+
+//! check if asynchronous (queue-ordered) memory buffers are supported by the given Accelerator
+template<typename TAcc>
+static constexpr auto isAsyncBufferSupported() -> bool
+{
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt>)
+    {
+        return (BOOST_LANG_CUDA >= BOOST_VERSION_NUMBER(11, 2, 0)) && (alpaka::Dim<TAcc>::value == 1);
+    }
+#endif // ALPAKA_ACC_GPU_CUDA_ENABLED
+
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevHipRt>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_GPU_HIP_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOacc>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOmp5>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+
+    return true;
+}
+
+
+// 0- and 1- dimensional space
+using Idx = std::size_t;
+using Dim0D = alpaka::DimInt<0u>;
+using Dim1D = alpaka::DimInt<1u>;
+using Scalar = alpaka::Vec<Dim0D, Idx>;
+
+// enabled accelerators with 1-dimensional kernel space
+using TestAccs = alpaka::test::EnabledAccs<Dim1D, Idx>;
+
+TEMPLATE_LIST_TEST_CASE("zeroDimBuffer", "[zeroDimBuffer]", TestAccs)
+{
+    using DeviceAcc = TestType;
+    using Device = alpaka::Dev<DeviceAcc>;
+    using DeviceQueue = alpaka::Queue<DeviceAcc, alpaka::NonBlocking>;
+
+    using HostAcc = alpaka::AccCpuSerial<Dim1D, Idx>;
+    using Host = alpaka::DevCpu;
+    using HostQueue = alpaka::Queue<HostAcc, alpaka::Blocking>;
+
+    // check that a Scalar extent has exactly 1 element
+    Scalar scalar;
+    INFO("Scalar size: " << scalar.prod())
+    CHECK(scalar.prod() == 1u);
+
+    // CPU host
+    auto const host = alpaka::getDevByIdx<Host>(0u);
+    INFO("Using alpaka accelerator: " << alpaka::getAccName<HostAcc>())
+    HostQueue hostQueue(host);
+
+    // host buffer
+    auto h_buffer1 = alpaka::allocBuf<int, Idx>(host, Scalar{});
+    INFO(
+        "host buffer allocated at " << alpaka::getPtrNative(h_buffer1) << " with "
+                                    << alpaka::extent::getExtentProduct(h_buffer1) << " element(s)")
+
+    // async host buffer
+    auto h_buffer2 = alpaka::allocAsyncBuf<int, Idx>(hostQueue, Scalar{});
```

Comment:
This is currently triggering a static assert within alpaka see [here](https://gitlab.com/hzdr/crp/alpaka/-/jobs/1897665119#L3637) because this feature needs CUDA 11.2.

The solution from line 142 should solve the issue:

```
auto d_buffer2 = isAsyncBufferSupported<DeviceAcc>() ? alpaka::allocAsyncBuf<int, Idx>(deviceQueue, Scalar{})
                                                         : alpaka::allocBuf<int, Idx>(device, Scalar{});
```

Reply:
Mhm, this is unexpected.
`hostQueue` refers to the CPU serial backend, for which `allocAsyncBuf` should always be defined.

Reply:
I think the specialization for zero-dimensional host allocation is missing and therefore is falling back to the device specialization.

Reply:
Mhh there should be no need for the specialization.

Reply:
I think I know the problem:
https://github.com/alpaka-group/alpaka/blob/5e008e48cf079c81593788713f12ed114291beaa/include/alpaka/mem/buf/BufUniformCudaHipRt.hpp#L342-L344

The compiler is allowed to evaluate the static assert within the class/struct at any time, not only if an instance of this object is created. The check must be deferred.

```
            static_assert(
                sizeof(TElem) != 0 && meta::DependentFalseType<TElem>::value,
                "Support for stream-ordered memory buffers requires CUDA 11.2 or higher.");
```

Checking the size of an template type will defere the check.

Reply:
Could you add this check as PR within your PR. I can on Monday open a separate PR we can to fix the error.

Reply:
Ok I missed that the assert is using a dependent fals, mmhh.

Reply:
I've setup a local test area with CUDA 11.0, and I can reproduce the error... investigating...

Reply:
I suspect it happens because both branches of the ternary `condition ? if_true : if_false` operator are compiled.